### ISSUE TITLE
SingleDatePicker: startDate and endDate passed to onClose when clicking outside

### ISSUE THREAD

**Issue by mauricesnip** [airbnb/react-dates#1674]
react-dates@20.2.0

While implementing the `SingleDatePicker` I found out that `onDayClick` passes `{ date: Moment }` and `onOutsideClick` passes `{ startDate: undefined, endDate: undefined }` to `onClose`. See: https://github.com/airbnb/react-dates/blob/11de2016bf6e7111ddebd5a7f7fc1fe929a3600c/src/components/SingleDatePicker.jsx#L206

Should the latter even be the case for `SingleDatePicker`, since there's only one date?

**Comment by ljharb** [comment]
This is a bug, since those two aren't [props of the SDP](https://github.com/airbnb/react-dates/blob/11de2016bf6e7111ddebd5a7f7fc1fe929a3600c/src/shapes/SingleDatePickerShape.js). 

These should be removed, and `date` should be passed instead.

**Comment by AlokTakshak** [comment]
@ljharb  i would like to work on that , we need to `remove` `startDate` and `endDate` and` pass`  only `date` to onClose right?

**Comment by ljharb** [comment]
Yes. Since they were only ever passed as undefined, it shouldn’t be a breaking change.

**Comment by AlokTakshak** [comment]
#1678

**Comment by manuelurenah** [comment]
My work around for this issue:
```jsx
<SingleDatePicker
  // ...
  onClose={date => {
    const parsed = JSON.parse(JSON.stringify(date));
    // Your date is at parsed.date
  }}
/>
```
